libraries/OPAMP/src/OPAMP.cpp

```diff
@@ -0,0 +1,69 @@
+#include "OPAMP.h"
+#include <Arduino.h>
+
+/* pin mode needed to activate OPAMP functionality */
+#define OPAMP_IN_PINCFG (IOPORT_CFG_PORT_DIRECTION_INPUT | IOPORT_CFG_PERIPHERAL_PIN | IOPORT_CFG_ANALOG_ENABLE)
+#define OPAMP_OUT_PINCFG (IOPORT_CFG_PORT_DIRECTION_OUTPUT | IOPORT_CFG_PERIPHERAL_PIN | IOPORT_CFG_ANALOG_ENABLE)
+#define FSP_CHECK(err) do { if( (err) != FSP_SUCCESS) return false; } while(0)
+
+bool OpampClass::initPins() {
+    fsp_err_t err;
+    ioport_instance_ctrl_t ioport_ctrl {};
+#if defined(ARDUINO_UNOWIFIR4) || defined(ARDUINO_MINIMA)
+    /* channel 0 pins. Only accessible ones. */
+    err = R_IOPORT_PinCfg(&ioport_ctrl, BSP_IO_PORT_00_PIN_00, OPAMP_IN_PINCFG);
+    FSP_CHECK(err);
+    err = R_IOPORT_PinCfg(&ioport_ctrl, BSP_IO_PORT_00_PIN_01, OPAMP_IN_PINCFG);
+    FSP_CHECK(err);
+    err = R_IOPORT_PinCfg(&ioport_ctrl, BSP_IO_PORT_00_PIN_02, OPAMP_OUT_PINCFG);
+    FSP_CHECK(err);
+#else
+#error "Unsupported board."
+#endif
+    // if we got here, none of the checks triggered an early return.
+    return true;
+}
+
+void OpampClass::initOpamp(OpampSpeedMode speed, uint8_t channel_mask) {
+    uint8_t ampmc_val  = 0U;
+    /* setup amplifier speed mode within amplifier mode control */
+    /* for all boards, this is at bit position 7 with either 0 (lowspeed) or 1 (highspeed) */
+    ampmc_val = (uint8_t) ((uint8_t) speed << R_OPAMP_AMPMC_AMPSP_Pos) & R_OPAMP_AMPMC_AMPSP_Msk;

```

Comment:
Loads of direct register manipulation in the sketch, normally we try to hide this within the library. Please get rid of all this from `.ino` and sensibly encapsulate it within the library.

Reply:
Darn this is in the library. Mea culpa. I take everything back.

Reply:
There are two ways to implement interaction with the OPAMP: Direct register manipulation or the [`R_OPAMP`](https://github.com/renesas/fsp/blob/v4.0.0/ra/fsp/src/r_opamp/r_opamp.c) library in the Renesas FSP framework (see https://github.com/maxgerhardt/uno-r4-opamp/tree/main/src). Since FSP is added as a precompiled library in this core I would have had to install Renesas e² studio, modify the config and recompile the `.a` static library files. Or, I could do register manipulation directly which should result in even smaller code even.

Just to make sure: Do you want me to use FSP APIs (which requires me to recompile the .a files) or is this direct writing into OPAMP registers okay?

Reply:
Direct register manipulation is okay :+1: (as long as its readable, currently it is).

Reply:
Then this thread can be resolved without action?

Reply:
Yes.